allennlp/training/metrics.py

```diff
@@ -0,0 +1,185 @@
+from typing import Dict, Optional
+
+from overrides import overrides
+import torch
+
+from allennlp.common.registrable import Registrable
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+
+
+class Metric(Registrable):
+    """
+    A very general abstract class representing a metric which can be
+    accumulated.
+    """
+    def __call__(self, predictions: torch.Tensor,
+                 gold_labels: torch.Tensor,
+                 mask: Optional[torch.Tensor]):
+
+        raise NotImplementedError
+
+    def get_metric(self, reset: bool) -> Dict[str, float]:
```

Comment:
I think this should return a float, not a dict, and the model should handle the dictionary keys.